py/ringbuf.c

```diff
@@ -83,17 +81,7 @@ int ringbuf_get_bytes(ringbuf_t *r, uint8_t *data, size_t data_len) {
     if (ringbuf_avail(r) < data_len) {
         return (r->size <= data_len) ? -2 : -1;
     }
-    uint32_t iget = r->iget;
-    uint32_t iget_a = (iget + data_len) % r->size;
-    uint8_t *datap = data;
-    if (iget_a < iget) {
-        // Copy part of the data from the space left at the end of the buffer
-        memcpy(datap, r->buf + iget, r->size - iget);
-        datap += (r->size - iget);
-        iget = 0;
-    }
-    memcpy(datap, r->buf + iget, iget_a - iget);
-    r->iget = iget_a;
+    ringbuf_memcpy_get_internal(r, data, data_len);
```

Comment:
We are going to need to add some tests that cover this.  Probably in `ports/unix/coverage.c`.

Reply:
Thanks for the suggestion, I've added some extra tests to the ringbuf section there. 
Even though these aren't my functions ;-) I did modify them, making this a good opportunity to improve overall coverage.

Reply:
Looks good now!